README.md

```diff
@@ -11,10 +11,28 @@ $ npm install openwhisk
 
 ## usage
 
+Instantiate openwhisk with specific options:
+
 ```
-var openwhisk = require('openwhisk');
-var options = {apihost: 'openwhisk.ng.bluemix.net', api_key: '...'};
-var ow = openwhisk(options);
+function main(params) {
+	// require the openwhisk npm package
+	var openwhisk = require('openwhisk');
+
+	// read apihost, auth, and namespace from params
+	var apiHost = params.apiHost; # e.g. openwhisk.ng.bluemix.net
+	var namespace = params.namespace;
+	var auth = params.auth; 
+
+	// generate api_key from auth
+	var base64Auth = new Buffer(auth).toString('base64');
+	var apiKey = "Basic " + base64Auth;
```

Comment:
Yup and found it weird (running on bluemix).

Without "return" on wsk.rules.list():
```
function main(params) {
    var openwhisk = require("openwhisk");

    var api_key = "user:pass";
    var base64_api_key = new Buffer(api_key).toString('base64');
    var encoded_api_key = "Basic " + base64_api_key;

    console.log(encoded_api_key)

    options = {
        api: "https://openwhisk.ng.bluemix.net/api/v1/",
        api_key: encoded_api_key,
        namespace: "my_namespace"
    }

    var wsk = openwhisk(options);

    wsk.rules.list()
    .then(function (packages) {
        console.log("PACKAGES");
        console.log(packages);
        return { packages: packages };
    })
    . catch(function (err) {
        console.log(err);
        return err;
    });
}
```
Result is:

 ```
   "response": {
        "status": "success",
        "statusCode": 0,
        "success": true,
        "result": {}
    },
```
With "return" wsk.rules.list():

```
function main(params) {
    var openwhisk = require("openwhisk");

    var api_key = "user:pass";
    var base64_api_key = new Buffer(api_key).toString('base64');
    var encoded_api_key = "Basic " + base64_api_key;

    console.log(encoded_api_key)

    options = {
        api: "https://openwhisk.ng.bluemix.net/api/v1/",
        api_key: encoded_api_key,
        namespace: "my_namespace"
    }

    var wsk = openwhisk(options);

    return wsk.rules.list()
    .then(function (packages) {
        console.log("PACKAGES");
        console.log(packages);
        return { packages: packages };
    })
    . catch(function (err) {
        console.log(err);
        return err;
    });
}
```
Result is:

```
  "response": {
        "status": "application error",
        "statusCode": 0,
        "success": false,
        "result": {
            "error": {
                "code": 3.11923e+06,
                "error": "The supplied authentication is invalid"
            }
        }
    },
```